src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -300,22 +301,15 @@ class AppsResource @Inject() (
   private def maybePostEvent(req: HttpServletRequest, app: AppDefinition) =
     eventBus.publish(ApiPostEvent(req.getRemoteAddr, req.getRequestURI, app))
 
-  private def search(cmd: String, id: String): Iterable[AppDefinition] = {
-    /* Returns true iff `a` is a prefix of `b`, case-insensitively */
-    def isPrefix(a: String, b: String): Boolean =
-      b.toLowerCase contains a.toLowerCase
+  private def search(cmd: Option[String], id: Option[String], label: Option[String]): Iterable[AppDefinition] = {
+    def containCaseInsensitive(a: String, b: String): Boolean = b.toLowerCase contains a.toLowerCase
+    val selectors = label.map(new LabelSelectorParsers().parsed)
 
     service.listApps().filter { app =>
-      val appMatchesCmd =
-        cmd != null &&
-          cmd.nonEmpty &&
-          app.cmd.exists(isPrefix(cmd, _))
-
-      val appMatchesId =
-        id != null &&
-          id.nonEmpty && isPrefix(id, app.id.toString)
-
-      appMatchesCmd || appMatchesId
+      val appMatchesCmd = cmd.fold(true)(c => app.cmd.exists(containCaseInsensitive(c, _)))
```

Comment:
The old code would only match, if there was some value. Than all the matcher were OR combined.  The mechanism here will match, if there is some value or no value. This result is AND combined. That generally changes the behavior - but as we agreed, it makes more sense...
I added some test cases to show this.